### PR TITLE
Added hyphens to make it in sync with the current writing style

### LINKED ISSUE #141: Replace some terms with their hyphenated version to make it in sync with the current style guide
According to current document writing style and [Google document style guide comprising word list](https://developers.google.com/style/word-list), we should rather use the hyphenated version of some terms like `Step by step` and `Lead in` which are present in `how-to/about-how-to.md` and `how-to/template-how-to.md` docs of the repo.

### PR DESCRIPTION
**Pull request by Dishebh**
Changes made:

- Added hyphens in phrases / words `Step by step` and `Lead in`.

Resolves #141

### CODE REVIEW

**Comment by Dishebh** [comment]
@jaredmorgs @camerons Please review :)

**Comment by camerons** [comment]
@Dishebh, thanks for this pull request. It looks likely that this is a valid improvement.
But, I can't see a reference to an issue that you are fixing. (I might be missing something, I'm not an expert on github).
It is good practice with all fixes to:
1. Create an issue.
2. Explain why you are making the fix, ideally referencing a document which backing up your reasoning. In this case, you'd be referencing a section in the Google Developers Style Guide.
3. Noting whether you have fixed it just in this document, or whether you have reviewed all templates for this rule you are fixing.
By doing this, you help the reviewers, (and lack of reviewers is often a bottleneck in getting pull requests completed.)
Cheers, Cameron

**Comment by Dishebh** [comment]
@camerons since it was a small change, I did not think of opening an issue for it. But your reasoning makes perfect sense, and how the whole process of issue -> discuss -> PR -> review -> merge makes it easier to track the changes in repo. I'll open an issue for it and refer it in the PR description.

File: how-to/template-how-to.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ Lead-in sentence for an ordered list:
 
 ### Step 3: Optional: title for step - code snippet
 
-Lead in sentence explaining the code snippet. E.g.: 
+Lead-in sentence explaining the code snippet. E.g.:
```
Comment (jaredmorgs) [inline how-to/template-how-to.md:39]: I know it is not part of the core issue here, but `E.g.:` is not recommended according to https://developers.google.com/style/abbreviations#dont-use

Could you please change this to:

"For example: "

Comment (Dishebh) [reply how-to/template-how-to.md]: Replaced abbreviation by its full-form. Glad that I could help @jaredmorgs :)

**Comment by shreyas1599** [comment]
@Dishebh perhaps you could edit this PR and others that you've opened and [link the relevant issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) so that they get closed automatically after merging.

**Comment by Dishebh** [comment]
@shreyas1599, sure. I edited this PR description and will make similar changes in my other PRs as well.

**Pull request merged**